### PR TITLE
Mam replace list

### PR DESCRIPTION
**Pull request by tsvisabo**
- General type container generator.bzl
- Provide type generic set implementation
- replace mammade list with new implemented set

### CODE REVIEW

File: utils/containers/set.h.tpl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  {TYPE} value;
+  UT_hash_handle hh;
+} {TYPE}_set_entry_t;
+
+typedef {TYPE}_set_entry_t *{TYPE}_set_t;
+
+typedef retcode_t (*{TYPE}_on_container_func)(void *container,
+                                                {TYPE} *type);
+
+uint32_t {TYPE}_set_size({TYPE}_set_t const *const set);
```
Comment (th0br0) [inline utils/containers/set.h.tpl]: this should be a `size_t`?

File: utils/containers/type_container_generator.bzl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "additional_deps": attr.string(mandatory = False),
+        "parent_directory": attr.string(mandatory = False),
+    },
+    outputs = {
+        "source": "%{source}",
+        "header": "%{header}",
+    },
+)
+
+def type_container_generate(container_type, value_type, additional_include_path = "", additional_deps = "//utils/containers:dummy_dep", parent_directory = "utils/containers"):
```
Comment (th0br0) [inline utils/containers/type_container_generator.bzl]: shouldn't this be `typed_container_generate`?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: utils/containers/set.h.tpl (hunk 1 above)
Comment (tsvisabo) [reply utils/containers/set.h.tpl]: sure

File: utils/containers/type_container_generator.bzl (hunk 2 above)
Comment (tsvisabo) [reply utils/containers/type_container_generator.bzl]: yeah, or maybe generic_container_generate?

Comment (tsvisabo) [reply utils/containers/type_container_generator.bzl]: done

File: utils/containers/set.h.tpl (hunk 1 above)
Comment (tsvisabo) [reply utils/containers/set.h.tpl]: done

**Review by th0br0 — DISMISSED** [review]

**Review by th0br0 — APPROVED** [review]

**Pull request merged**